main.tf

```
@@ -17,7 +17,8 @@ locals {
 resource "aws_efs_file_system" "default" {
   count                           = var.enabled ? 1 : 0
   tags                            = module.label.tags
-  encrypted                       = var.encrypted
+  encrypted                       = var.encrypted || var.kms_key_id ? true : false
```

```suggestion
  encrypted                       = var.encrypted
```

I would not use the logic to calculate if it needs to be encrypted or not.
If you want it to be encrypted, set `var.encrypted` to `true`.
If you want it to be encrypted and want to use an existing key, set `var.encrypted` to `true` and provide `var.kms_key_id`.
Easier to understand and fewer surprises.

thanks @dhutty